packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Tabs/tabs.js

```diff
@@ -0,0 +1,297 @@
+import React, {
+    Children,
+    useRef,
+    useEffect,
+    useCallback,
+    useState
+} from 'react';
+import {
+    Tabs as TabWrapper,
+    TabList,
+    Tab as TabHeader,
+    TabPanel
+} from 'react-tabs';
+import defaultClasses from './tabs.css';
+import { mergeClasses } from '../../../../../classify';
+import { arrayOf, number, oneOf, shape, string } from 'prop-types';
+
+/**
+ * Page Builder Tabs component.
+ *
+ * This component is part of the Page Builder / PWA integration. It can be consumed without Page Builder.
+ *
+ * @typedef Tabs
+ * @kind functional component
+ *
+ * @param {props} props React component props
+ *
+ * @returns {React.Element} A React component that displays a set of Tabs.
+ */
+const Tabs = props => {
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const navigationRef = useRef(null);
+    const [isScrolling, setIsScrolling] = useState(false);
+    const [clientX, setClientX] = useState(0);
+    const [scrollX, setScrollX] = useState(0);
+    const [scrollElement, setScrollElement] = useState(null);
+    const [navWrapperClass, setNavWrapperClass] = useState();
+    const {
+        tabNavigationAlignment = 'left',
+        minHeight,
+        defaultIndex = 0,
+        headers = [],
+        textAlign,
+        border,
+        borderColor,
+        borderWidth,
+        borderRadius,
+        marginTop,
+        marginRight,
+        marginBottom,
+        marginLeft,
+        paddingTop,
+        paddingRight,
+        paddingBottom,
+        paddingLeft,
+        cssClasses = [],
+        children
+    } = props;
+
+    const onMouseDown = useCallback(
+        event => {
+            setIsScrolling(true);
+            setClientX(event.clientX);
+        },
+        [setIsScrolling, setClientX]
+    );
+
+    const onMouseUp = useCallback(() => {
+        setIsScrolling(false);
+    }, [setIsScrolling]);
+
+    const onMouseMove = useCallback(
+        event => {
+            if (isScrolling && scrollElement) {
+                scrollElement.scrollLeft = scrollX + (clientX - event.clientX);
+                setScrollX(scrollElement.scrollLeft);
+                setClientX(event.clientX);
+            }
```

Comment:
Setting state and triggering a render on `mousemove` is generally unwise. This event fires rapidly and reconciliation is not that fast, especially for nontrivial components such as this one, and even less so if you have side effects on each render (which this one does).

Consider throttling this event, or perhaps constraining its updates to a ref and then using an effect to sync them with the DOM on an interval.

Reply:
Oh daaamnn, yeah I did a test and this was causing the component to get rendered a lot. I've updated this to use `useRef` and now it doesn't re-render 5000 times.
https://github.com/magento/pwa-studio/pull/1872/commits/dc8fb48bc7c2edba5bd3573da6e9fa28e48adfc1